src/Razor/src/Microsoft.VisualStudio.RazorExtension/Options/AdvancedOptionPage.cs

```diff
@@ -31,7 +34,35 @@ public AdvancedOptionPage()
     [LocDisplayName(nameof(VSPackage.Setting_FormattingOnTypeDisplayName))]
     public bool FormatOnType
     {
-        get => _optionsStorage.Value.FormatOnType;
-        set => _optionsStorage.Value.FormatOnType = value;
+        get => _formatOnType ?? _optionsStorage.Value.FormatOnType;
+        set => _formatOnType = value;
+    }
+
+    [LocCategory(nameof(VSPackage.Typing))]
+    [LocDescription(nameof(VSPackage.Setting_AutoClosingTagsDescription))]
+    [LocDisplayName(nameof(VSPackage.Setting_AutoClosingTagsDisplayName))]
+    public bool AutoClosingTags
+    {
+        get => _autoClosingTags ?? _optionsStorage.Value.AutoClosingTags;
+        set => _autoClosingTags = value;
+    }
+
+    protected override void OnApply(PageApplyEventArgs e)
```

Comment:
Good catch on this. I was realizing that we weren't following the apply/cancel behavior. Thanks for fixing!